### PR TITLE
Partial derivatives (of the same coordinate system) commute

### PR DESCRIPTION
**Pull request by unalmis**
A few compute functions do a lot of work to compute the same thing as another, sometimes with slightly different math. It's better to just alias them to the other so that there is less math to check as mentioned  [here](https://github.com/PlasmaControl/DESC/pull/648#issuecomment-1718219257).

### CODE REVIEW

File: desc/compute/_basis_vectors.py [hunk 1]
```diff
@@ -1811,59 +1811,43 @@ def _e_sub_rho_rrt(params, transforms, profiles, data, **kwargs):
 def _e_sub_rho_rrz(params, transforms, profiles, data, **kwargs):
     data["e_rho_rrz"] = jnp.array(
         [
-            -2 * data["omega_rz"] * data["R_r"] * data["omega_r"]
```
Comment (unalmis) [inline desc/compute/_basis_vectors.py:1814]: this is the same as the other existing mixed partial; there's nothing to check here just copy pasted

**Review by dpanici — CHANGES_REQUESTED** [review]
Good catch, @kianorr has previously implemented an `alias` keyword to the decorator just for this purpose, can you remove the duplicate compute definitions and use the `alias` kwarg to accomplish this instead?

example is here: https://github.com/PlasmaControl/DESC/blob/f7383817d9aee40d54fec1d9a36e685d0cfe7c0c/desc/compute/_metric.py#L1373

**Comment by dpanici** [comment]
Also maybe in `test_compute_everything` we also compute the aliases? or would that be too much duplicated computation? I am fine either way, as of now this PR works and everything is also tested bc of that test, but with the `alias` method, there won't be a test that the aliased things are computed correctly, idt.

**Comment by unalmis** [comment]
> Also maybe in `test_compute_everything` we also compute the aliases? or would that be too much duplicated computation? I am fine either way, as of now this PR works and everything is also tested bc of that test, but with the `alias` method, there won't be a test that the aliased things are computed correctly, idt.

i don't think this is worth testing

File: desc/compute/_basis_vectors.py [hunk 2]
```diff
@@ -1969,6 +1955,7 @@ def _e_sub_rho_rt(params, transforms, profiles, data, **kwargs):
         "desc.geometry.surface.ZernikeRZToroidalSection",
     ],
     basis="basis",
+    aliases=["e_theta_rrt"],
```
Comment (unalmis) [inline desc/compute/_basis_vectors.py:1958]: permutations like e_theta_rtr are connected automatically

**Comment by unalmis** [comment]
If any of the quantities with an alias in the list of variables [here](https://desc-docs--1068.org.readthedocs.build/en/1068/variables.html) can be discontinuous, then we should undo commit d1863062f4ce83008e29a79e564a9e65eb00688e before merging. (The basis vectors are always continuous, so the jacobian derivatives etc. are too - and i don't think desc can get discontinuous derivatives of |B|)

**Comment by kianorr** [comment]
> If any of the quantities with an alias in the list of variables [here](https://desc-docs--1068.org.readthedocs.build/en/1068/variables.html) can be discontinuous, then we should undo commit [d186306](https://github.com/PlasmaControl/DESC/commit/d1863062f4ce83008e29a79e564a9e65eb00688e) before merging. (The basis vectors are always continuous, so the jacobian derivatives etc. are too - and i don't think desc can get discontinuous derivatives of |B|)

what do you mean by discontinuous/continuous?

**Comment by unalmis** [comment]
> > If any of the quantities with an alias in the list of variables [here](https://desc-docs--1068.org.readthedocs.build/en/1068/variables.html) can be discontinuous, then we should undo commit [d186306](https://github.com/PlasmaControl/DESC/commit/d1863062f4ce83008e29a79e564a9e65eb00688e) before merging. (The basis vectors are always continuous, so the jacobian derivatives etc. are too - and i don't think desc can get discontinuous derivatives of |B|)
> 
> what do you mean by discontinuous/continuous?

For example, the components of B tangential to a surface current will be discontinuous. It's not as obvious whether interchanging partial derivatives is valid near discontinuities, and one would have to do math or seek weaker conditions to justify it. I don't think we need to worry about this - all the derivatives of B are compositions of continuous functions as we have implemented. I was mainly just documenting the assumptions for commit [d186306](https://github.com/PlasmaControl/DESC/commit/d1863062f4ce83008e29a79e564a9e65eb00688e) in case anyone changes stuff in future.

**Review by dpanici — APPROVED** [review]

**Review by ddudt — APPROVED** [review]

**Pull request merged**